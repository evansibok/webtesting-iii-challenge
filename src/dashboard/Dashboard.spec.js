// Test away
import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'
import Dashboard from './Dashboard'

afterEach(rtl.cleanup);

let wrapper;
beforeEach(() => {
  wrapper = rtl.render(<Dashboard />)
});

const unlocked = () => wrapper.queryByText(/Unlocked/i);
const open = () => wrapper.queryByText(/Open/i);
const lockGate = () => wrapper.queryByText(/Lock Gate/i);
const closeGate = () => wrapper.queryByText(/Close Gate/i);
const closed = () => wrapper.queryByText(/Closed/i);
const openGate = () => wrapper.queryByText(/Open Gate/i);
const locked = () => wrapper.queryByText(/Locked/i);
const unlockGate = () => wrapper.queryByText(/Unlock Gate/i);


describe('Dashboard Component renders without crashing', () => {
  it('matches previous snapshot', () => {
    expect(wrapper.container).toMatchSnapshot();
  })

  it('displays initial components correctly', () => {
    expect(unlocked()).toBeInTheDocument();
    expect(open()).toBeInTheDocument();
    expect(lockGate()).toBeInTheDocument();
    expect(closeGate()).toBeInTheDocument();
    expect(lockGate()).toBeDisabled();
  })
})

describe('Close Gate works properly', () => {

  it('on "Close Gate" click "Open" changes to "Closed"', () => {
    rtl.fireEvent.click(closeGate());
    expect(closed()).toBeInTheDocument();
    expect(openGate()).toBeInTheDocument();
    expect(lockGate()).not.toBeDisabled();
    expect(wrapper.container).toMatchSnapshot();
  })
})
