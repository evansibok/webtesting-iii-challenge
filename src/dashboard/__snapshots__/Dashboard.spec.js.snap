// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Close Gate works properly on "Close Gate" click "Open" changes to "Closed" 1`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led red-led"
    >
      Closed
    </div>
  </div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
    >
      Open Gate
    </button>
  </div>
</div>
`;

exports[`Dashboard Component renders without crashing matches previous snapshot 1`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
    >
      Open
    </div>
  </div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
    >
      Close Gate
    </button>
  </div>
</div>
`;
